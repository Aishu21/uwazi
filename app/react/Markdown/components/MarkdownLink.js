import { Link } from 'react-router';
import PropTypes from 'prop-types';
import React from 'react';

const MarkdownLink = ({ url, className, children }) => (
  <Link to={url} className={className} href={url}>{children}</Link>
);

MarkdownLink.defaultProps = {
  children: '',
  className: '',
};

MarkdownLink.propTypes = {
  url: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]),
};

export default MarkdownLink;
