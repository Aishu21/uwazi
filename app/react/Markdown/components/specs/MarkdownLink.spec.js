import React from 'react';

import { shallow } from 'enzyme';

import MarkdownLink from '../MarkdownLink.js';

describe('Link', () => {
  it('should render a react-router Link when the url is relative', () => {
    const component = shallow(<MarkdownLink url="url" label="label" />);
    expect(component).toMatchSnapshot();
  });

  describe('when passing an icon property', () => {
    it('should render an extra div with icon class', () => {
      const component = shallow(<MarkdownLink url="url" label="label" icon="iconName"/>);
      expect(component).toMatchSnapshot();
    });

    describe('when passing a className property', () => {
      it('should append the className for the wrapping div', () => {
        const component = shallow(<MarkdownLink url="url" label="label" icon="iconName" className="extraClassName"/>);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
