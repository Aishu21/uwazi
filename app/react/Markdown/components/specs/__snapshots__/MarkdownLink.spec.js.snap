// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render a react-router Link when the url is relative 1`] = `
<Link
  href="url"
  onlyActiveOnIndex={false}
  style={Object {}}
  to="url"
>
  label
</Link>
`;

exports[`Link when passing an icon property should render an extra div with icon class 1`] = `
<div
  className="icon-link "
>
  <Link
    href="url"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="url"
  >
    <div>
      <i
        className="fa fa-iconName"
      />
    </div>
    <span>
      label
    </span>
  </Link>
</div>
`;

exports[`Link when passing an icon property when passing a className property should append the className for the wrapping div 1`] = `
<div
  className="icon-link extraClassName"
>
  <Link
    href="url"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="url"
  >
    <div>
      <i
        className="fa fa-iconName"
      />
    </div>
    <span>
      label
    </span>
  </Link>
</div>
`;
